src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './CodeSnippet.css';
+
+class CodeSnippet extends React.Component {
+  render() {
+    return (
+      <div className="p-code-snippet">
+        <input className="p-code-snippet__input" value={this.props.value} readOnly="readonly" />
+        <button className="p-code-snippet__action">Copy to clipboard</button>
```

Comment:
These components should be functional as they are written in js. Please adopt the functionality from this snippet: https://pastebin.canonical.com/202016/.

There should be a click listener on both the button and input.